code/game/objects/items/tools/welder.dm

```diff
@@ -100,9 +100,9 @@
 		return FALSE
 
 // When welding is about to start, run a normal tool_use_check, then flash a mob if it succeeds.
-/obj/item/weldingtool/tool_start_check(mob/living/user, amount=0)
+/obj/item/weldingtool/tool_start_check(atom/target, mob/living/user, amount=0)
 	. = tool_use_check(user, amount)
-	if(. && user)
+	if(. && user && target != user)
```

Comment:
Leaving this as reminder. Apparently people are getting flashed from other people's robo-limbs and also from repairing borgs. Instead of what I wrote here, this should probably have a check for `!issythetic(target)` or whatever. It's a snowflakey method, but maybe there's a better way